### PR TITLE
feat(condo): DOMA-5920 NewsItemCard page

### PR DESCRIPTION
**Pull request by VKislov**
image
image

### CODE REVIEW

File: apps/condo/pages/news/[id]/index.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+            case NEWS_TYPE_COMMON: {
+                newsItemType = Regular
+                break
+            }
+            case NEWS_TYPE_EMERGENCY: {
+                newsItemType = Emergency
+                break
+            }
+        }
+    }
```
Comment (AleX83Xpert) [inline apps/condo/pages/news/[id]/index.tsx:98]:
```suggestion
    const typesNamesMapping = {
        [NEWS_TYPE_COMMON]: Regular,
        [NEWS_TYPE_EMERGENCY]: Emergency,
    }
    const newsItemType = typesNamesMapping[get(newsItem, 'type')] || ''
```

File: apps/condo/pages/news/[id]/index.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    const createdBy = get(newsItem, 'createdBy.id')
+    const {
+        obj: employee,
+        loading: employeeLoading,
+        error: employeeError,
+
```
Comment (AleX83Xpert) [inline apps/condo/pages/news/[id]/index.tsx:104]:
```suggestion
```

File: apps/condo/pages/news/[id]/index.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                                <DeleteButtonWithConfirmModal
+                                    key='delete'
+                                    title={ConfirmDeleteTitle}
+                                    message={ConfirmDeleteMessage}
+                                    okButtonLabel={DeleteTitle}
+                                    action={handleDeleteButtonClick}
+                                    buttonContent={DeleteTitle}
+                                />,
+                            ]}/>)
+                        }
```
Comment (AleX83Xpert) [inline apps/condo/pages/news/[id]/index.tsx:198]:
```suggestion
                        {get(newsItem, 'sentAt') ? (
                            <Link key='update' href={`/news/${get(newsItem, 'id')}/update`}>
                                <Button
                                    type='primary'
                                >
                                    {ResendTitle}
                                </Button>
                            </Link>
                            ) : ( <ActionBar actions={[
                                <Link key='update' href={`/news/${get(newsItem, 'id')}/update`}>
                                    <Button
                                        type='primary'
                                    >
                                        {EditTitle}
                                    </Button>
                                </Link>,
                                <DeleteButtonWithConfirmModal
                                    key='delete'
                                    title={ConfirmDeleteTitle}
                                    message={ConfirmDeleteMessage}
                                    okButtonLabel={DeleteTitle}
                                    action={handleDeleteButtonClick}
                                    buttonContent={DeleteTitle}
                                />,
                            ]}/>)
                        }
```

File: apps/condo/pages/news/[id]/index.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        fieldValue,
+    } = props
+    return (
+        <>
+            <Col span={8}>
+                <Typography.Text type='secondary'>
+                    {fieldTitle}
+                </Typography.Text>
+            </Col>
+            <Col span={16} style={{ width: '100%' }}>
```
Comment (Alllex202) [inline apps/condo/pages/news/[id]/index.tsx:46]: you should move style to const

File: apps/condo/pages/news/[id]/index.tsx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        createdBy: get(employee, 'name'),
+        isOwner: createdBy === user.id,
+    })
+
+    const isLoading = employeeLoading || newsItemLoading
+    const hasError = employeeError || newsItemError
+    const isNotFound = !isLoading && (!employee || !newsItem)
+    if (hasError || isLoading || isNotFound) {
+        const errorToPrint = hasError ? ServerErrorMsg : isNotFound ? NotFoundMsg : null
+        return <LoadingOrErrorPage title='' loading={isLoading} error={errorToPrint}/>
```
Comment (Alllex202) [inline apps/condo/pages/news/[id]/index.tsx:122]:
```suggestion
        return <LoadingOrErrorPage loading={isLoading} error={errorToPrint}/>
```

File: apps/condo/pages/news/[id]/index.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+    }, [softDeleteNews, newsItem])
+
+    const CreatedByLabel = intl.formatMessage({ id: 'pages.news.newsItemCard.author' }, {
+        createdBy: get(employee, 'name'),
+        isOwner: createdBy === user.id,
+    })
+
+    const isLoading = employeeLoading || newsItemLoading
+    const hasError = employeeError || newsItemError
+    const isNotFound = !isLoading && (!employee || !newsItem)
```
Comment (Alllex202) [inline apps/condo/pages/news/[id]/index.tsx:119]: Do you definitely want to not display the news item if for some reason the employee was not loaded, and the news item itself was loaded at the same time?
Employee used only for `CreatedByLabel`

File: apps/condo/pages/news/[id]/index.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+                >
+                    <Col span={24}>
+                        <PageHeader title={<Typography.Title>{PageTitleMsg}</Typography.Title>} />
+                        <Typography.Text type='secondary'>
+                            {CreatedByLabel}
+                        </Typography.Text>
+                    </Col>
+                    <Col span={24}>
+                        <FrontLayerContainer>
+                            <Row gutter={[0, 24]}>
```
Comment (Alllex202) [inline apps/condo/pages/news/[id]/index.tsx:139]: Should move gutter to const

File: apps/condo/pages/news/[id]/index.tsx [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+import { NEWS_TYPE_COMMON, NEWS_TYPE_EMERGENCY } from '@condo/domains/news/constants/newsTypes'
+import { NewsItem } from '@condo/domains/news/utils/clientSchema'
+import { OrganizationRequired } from '@condo/domains/organization/components/OrganizationRequired'
+import { OrganizationEmployee } from '@condo/domains/organization/utils/clientSchema'
+import { NotDefinedField } from '@condo/domains/user/components/NotDefinedField'
+
+
+const PAGE_ROW_GUTTER: RowProps['gutter'] = [0, 40]
+
+const FieldPairRow = (props) => {
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply apps/condo/pages/news/[id]/index.tsx:34]: Something like
```suggestion
interface IFieldPairRowProps {
    fieldTitle: string,
    fieldValue: string,
}

const FieldPairRow: React.FC<IFieldPairRowProps> = (props) => {
```

**Review by AleX83Xpert — APPROVED** [review]

File: apps/condo/pages/news/[id]/index.tsx (hunk 4 above)
Comment (VKislov) [reply apps/condo/pages/news/[id]/index.tsx:50]: But what's the point of moving this to a constant? It is not used anywhere else and moving to a constant does not add readability in this case.

Comment (AleX83Xpert) [reply apps/condo/pages/news/[id]/index.tsx:51]: Just a micro-optimization and best practices, to prevent creating a new object every render.

**Pull request merged**